tool/src/org/antlr/v4/codegen/UnicodeEscapes.java

```diff
@@ -8,31 +8,42 @@
 
 /**
  * Utility class to escape Unicode code points using various
- * languages' syntaxes.
+ * languages' syntax.
  */
-public abstract class UnicodeEscapes {
-	static public void appendJavaStyleEscapedCodePoint(int codePoint, StringBuilder sb) {
-		if (Character.isSupplementaryCodePoint(codePoint)) {
-			// char is not an 'integral' type, so we have to explicitly convert
-			// to int before passing to the %X formatter or else it throws.
-			sb.append(String.format("\\u%04X", (int)Character.highSurrogate(codePoint)));
-			sb.append(String.format("\\u%04X", (int)Character.lowSurrogate(codePoint)));
-		}
-		else {
-			sb.append(String.format("\\u%04X", codePoint));
-		}
+public class UnicodeEscapes {
+	public static String escapeCodePoint(int codePoint, TargetType targetType) {
+		StringBuilder result = new StringBuilder();
+		appendEscapedCodePoint(result, codePoint, targetType);
+		return result.toString();
 	}
 
-	static public void appendPythonStyleEscapedCodePoint(int codePoint, StringBuilder sb) {
-		if (Character.isSupplementaryCodePoint(codePoint)) {
-			sb.append(String.format("\\U%08X", codePoint));
-		}
-		else {
-			sb.append(String.format("\\u%04X", codePoint));
+	public static void appendEscapedCodePoint(StringBuilder sb, int codePoint, TargetType targetType) {
+		switch (targetType) {
+			case Java:
+			case JavaScript:
+			case Dart:
+				if (Character.isSupplementaryCodePoint(codePoint)) {
+					// char is not an 'integral' type, so we have to explicitly convert
+					// to int before passing to the %X formatter or else it throws.
+					sb.append(String.format("\\u%04X", (int)Character.highSurrogate(codePoint)));
+					sb.append(String.format("\\u%04X", (int)Character.lowSurrogate(codePoint)));
+				}
+				else {
+					sb.append(String.format("\\u%04X", codePoint));
+				}
+				break;
+			case CSharp:
+			case Python2:
+			case Python3:
+			case Cpp:
+			case Go:
+			case PHP:
+				String format = Character.isSupplementaryCodePoint(codePoint) ? "\\U%08X" : "\\u%04X";
+				sb.append(String.format(format, codePoint));
+				break;
+			case Swift:
+				sb.append(String.format("\\u{%04X}", codePoint));
+				break;
```

Comment:
Probably we also need `default` branch here for unofficial targets (use `Java` implementation or throw an exception).